src/regex.c

```diff
@@ -286,19 +287,22 @@ int do_regex_hybrid_crack(struct db_main *db, const char *regex,
 	}
 
 	while (c_iterator_next(iter)) {
-		c_iterator_value(iter, buffer);
-		c_simplestring_to_utf8_string(buffer, &word[0], sizeof(word));
 		c_simplestring_clear(buffer);
-		if (options.internal_cp != UTF_8)
-			utf8_to_cp_r(word, word, sizeof(word));
+		c_iterator_value(iter, buffer);
+		word = c_simplestring_to_string(buffer);
+    /**
+      * rexgen already creates the correct encoding
+      */
+		//if (options.internal_cp != UTF_8)
+		//	utf8_to_cp_r(word, word, sizeof(word));
 		if (options.mask) {
```

Comment:
Yes, they did, but conversion is slow. As rexgen does not really parse single characters in most cases, the actual locale doesn't matter (with one exception). Therefore, I handle Strings not as list of characters, but as list of bytes.

We only should do a conversion if necessary; i.e. if input locale is different from output locale. If you use CP1252 internally and configured the locale accordingly, than rexgen uses CP1252 as well.

If the output locale is different from the input locale, than rexgen should use the output locale as internal encoding (for performance reasons)